scripts/configuration.md

```diff
@@ -0,0 +1,194 @@
+The sections below describe each safety-results-over-time setting as of version 2.2.0.
+
+# settings.id_col
+`string`
+
+unique identifier variable name
+
+**default:** `"USUBJID"`
+
+
+
+# settings.time_settings
+`object`
+
+visit metadata
+
+## settings.time_settings.value_col
+`string`
+
+Visit variable name
+
+**default:** `"VISIT"`
+
+## settings.time_settings.label
+`string`
+
+Visit variable label
+
+**default:** `"Visit"`
+
+## settings.time_settings.order_col
+`string`
+
+Visit ordering variable name
+
+**default:** `"VISITNUM"`
+
+## settings.time_settings.order
+`array`
+
+Visit order
+
+**default:** none
+
+## settings.time_settings.rotate_tick_labels
+`boolean`
+
+Rotate tick labels 45 degrees?
+
+**default:** `"true"`
+
+## settings.time_settings.vertical_space
+`number`
+
+Rotated tick label spacing
+
+**default:** `"100"`
+
+
+
+# settings.measure_col
+`string`
+
+measure variable name
+
+**default:** `"TEST"`
+
+
+
+# settings.unit_col
+`string`
+
+measure unit variable name
+
+**default:** `"STRESU"`
+
+
+
+# settings.value_col
+`string`
+
+result variable name
+
+**default:** `"STRESN"`
+
+
+
+# settings.normal_col_low
+`string`
+
+LLN variable name
+
+**default:** `"STNRLO"`
+
+
+
+# settings.normal_col_high
+`string`
+
+ULN variable name
+
+**default:** `"STNRHI"`
+
+
+
+# settings.start_value
+`string`
+
+value of measure to display initially
+
+**default:** none
+
+
+
+# settings.filters
+`array`
+
+an array of filter variables and associated metadata
+
+**default:** none
+
+## settings.filters[].value_col
+`string`
+
+Variable name
+
+## settings.filters[].label
+`string`
+
+Variable label
+
+
+
+# settings.groups
+`array`
+
+an array of grouping variables and associated metadata
+
+**default:** none
+
+## settings.groups[].value_col
+`string`
+
+Variable name
+
+## settings.groups[].label
+`string`
+
+Variable label
+
+
+
+# settings.boxplots
+`boolean`
+
+controls initial display of box plots
+
+**default:** `"true"`
+
+
+
+# settings.violins
+`boolean`
+
+controls initial display of violin plots
+
+**default:** none
+
+
+
+# settings.unscheduled_visits
+`boolean`
+
+controls initial display of unscheduled visits
+
+**default:** none
+
+
+
+# settings.unscheduled_visit_pattern
+`string`
+
```

Comment:
is this a string? or a regular expression? (maybe json schema only supports strings?)

Reply:
nevermind - i see the logic below dealing with this.